query/sort.js

```diff
@@ -4,47 +4,65 @@ var popularity = 'popularity';
 var category = 'category';
 var category_weights = require('../helper/category_weights');
 var weights = require('pelias-suggester-pipeline').weights;
+var isObject = require( 'is-object' );
 
-module.exports = [
-  {
-    '_script': {
-      'file': admin_boost,
-      'type': 'number',
-      'order': 'desc'
-    }
-  },
-  {
-    '_script': {
-      'file': population,
-      'type': 'number',
-      'order': 'desc'
-    }
-  },
-  {
-    '_script': {
-      'file': popularity,
-      'type': 'number',
-      'order': 'desc'
-    }
-  },
-  {
-    '_script': {
-      'params': {
-        'category_weights': category_weights
-      },
-      'file': category,
-      'type': 'number',
-      'order': 'desc'
-    }
-  },
-  {
-    '_script': {
-      'params': {
-        'weights': weights
-      },
-      'file': 'weights',
-      'type': 'number',
-      'order': 'desc'
+module.exports = function( params ){
+  var scriptsConfig = [
+    {
+      '_script': {
+        'file': admin_boost,
+        'type': 'number',
+        'order': 'desc'
+      }
+    },
+    {
+      '_script': {
+        'file': population,
+        'type': 'number',
+        'order': 'desc'
+      }
+    },
+    {
+      '_script': {
+        'file': popularity,
+        'type': 'number',
+        'order': 'desc'
+      }
+    },
+    {
+      '_script': {
+        'params': {
+          'category_weights': category_weights
+        },
+        'file': category,
+        'type': 'number',
+        'order': 'desc'
+      }
+    },
+    {
+      '_script': {
+        'params': {
+          'weights': weights
+        },
+        'file': 'weights',
+        'type': 'number',
+        'order': 'desc'
+      }
     }
+  ];
+
+  if( isObject( params ) && params.hasOwnProperty( 'input' ) ){
+    scriptsConfig.push({
+      _script: {
```

Comment:
Yeah, they shouldn't have been there in the first place. I'd like to avoid doing that in this PR since it'll clutter the diff and is an unrelated cleanup change, though; will gladly do it in another branch/directly in `master`.